packages/medusa/src/api/store/products/helpers.ts

```diff
@@ -30,3 +47,101 @@ export const maybeApplyStockLocationId = async (req: MedusaRequest, ctx) => {
 
   return entities.map((entity) => entity.stock_location_id)
 }
+
+export const wrapProductsWithTaxPrices = async <T>(
+  req: RequestWithContext<T>,
+  products: HttpTypes.StoreProduct[]
+) => {
+  // If we are missing the necessary context, we can't calculate the tax, so only `calculated_amount` will be available
+  if (
+    !req.taxContext?.taxInclusivityContext ||
+    !req.taxContext?.taxLineContext
+  ) {
+    return
+  }
+
+  // If automatic taxes are not enabled, we should skip calculating any tax
+  if (!req.taxContext.taxInclusivityContext.automaticTaxes) {
+    return
+  }
+
+  const { isTaxInclusive } = req.taxContext.taxInclusivityContext
+  const taxService = req.scope.resolve(ModuleRegistrationName.TAX)
+
+  const taxRates = (await taxService.getTaxLines(
+    products.map(asTaxItem).flat(),
+    req.taxContext.taxLineContext
+  )) as unknown as ItemTaxLineDTO[]
+
+  const taxRatesAsMap = new Map<string, ItemTaxLineDTO[]>()
+  taxRates.forEach((taxRate) => {
+    if (!taxRatesAsMap.has(taxRate.line_item_id)) {
+      taxRatesAsMap.set(taxRate.line_item_id, [])
+    }
+
+    taxRatesAsMap.get(taxRate.line_item_id)?.push(taxRate)
+  })
+
+  products.forEach((product) => {
+    product.variants?.forEach((variant) => {
+      const taxRatesForVariant = taxRatesAsMap.get(variant.id) || []
+      const { priceWithTax, priceWithoutTax } = getTaxPricesForVariant(
+        variant,
+        taxRatesForVariant,
+        isTaxInclusive
+      )
+
+      ;(variant.calculated_price as any).calculated_amount_with_tax =
+        priceWithTax
+      ;(variant.calculated_price as any).calculated_amount_without_tax =
+        priceWithoutTax
+    })
+  })
+}
+
+const getTaxPricesForVariant = (
+  variant: HttpTypes.StoreProductVariant,
+  taxRatesForVariant: ItemTaxLineDTO[],
+  isTaxInclusive: boolean
+) => {
+  const tax = calculateTaxTotal({
+    // TODO: See why rate is nullable in tax
+    taxLines: taxRatesForVariant as any,
+    taxableAmount: variant.calculated_price!.calculated_amount!,
+    includesTax: isTaxInclusive,
+  })
+
+  const priceWithoutTax = isTaxInclusive
+    ? tax.minus(variant.calculated_price!.calculated_amount!).abs().toNumber()
```

Comment:
Wouldn't it be more human readable to do `price - tax` instead of `absolute(tax - price)`?

Reply:
Yeah it is more readable, it's just that calculated amount is not a big number here, so instead of converting I did this. Let me revisit it though, I agree with you